extension/json/json_common.cpp

```diff
@@ -0,0 +1,135 @@
+#include "json_common.hpp"
+
+namespace duckdb {
+
+static void CheckPath(const Value &path_val, string &path, size_t &len) {
+	string error;
+	Value path_str_val;
+	if (!path_val.TryCastAs(LogicalType::VARCHAR, path_str_val, &error)) {
+		throw InvalidInputException("Cannot cast JSON path to VARCHAR");
+	}
+	auto path_str = path_str_val.GetValueUnsafe<string_t>();
+	len = path_str.GetSize();
+	auto ptr = path_str.GetDataUnsafe();
+	// Empty strings and invalid $ paths yield an error
+	if (len == 0 || (*ptr == '$' && !JSONCommon::ValidPathDollar(ptr, len))) {
+		throw Exception("JSON path error");
```

Comment:
Could we get a more descriptive error message perhaps, e.g. `JSONCommon::ValidPathDollar` could write to an error message parameter?